src/main/java/org/apache/maven/plugins/ejb/EjbMojo.java

```diff
@@ -465,10 +465,10 @@ private File generateEjbClient()
     private void checkEJBVersionCompliance( File deploymentDescriptor )
         throws MojoExecutionException
     {
-        if ( !ejbVersion.matches( "\\A[2-3]\\.[0-9]\\z" ) )
+        if ( !ejbVersion.matches( "\\A[2-4]\\.[0-9]\\z" ) )
```

Comment:
a unit test would be easier to manage than an integration test

Reply:
> a unit test would be easier to manage than an integration test

IT replaced with UTs.